agroal-pool/src/main/java/io/agroal/pool/wrapper/XAResourceWrapper.java

```diff
@@ -0,0 +1,121 @@
+package io.agroal.pool.wrapper;
+
+import io.agroal.pool.ConnectionHandler;
+import io.agroal.pool.util.AutoCloseableElement;
+
+import javax.transaction.xa.XAException;
+import javax.transaction.xa.XAResource;
+import javax.transaction.xa.Xid;
+import java.lang.reflect.InvocationHandler;
+import java.sql.SQLException;
+
+import static java.lang.reflect.Proxy.newProxyInstance;
+
+/**
+ * @author <a href="lbarreiro@redhat.com">Luis Barreiro</a>
+ */
+public class XAResourceWrapper extends AutoCloseableElement implements XAResource {
+
+    private static final String CLOSED_HANDLER_STRING = XAConnectionWrapper.class.getSimpleName() + ".CLOSED_XA_RESOURCE";
+
+    private static final InvocationHandler CLOSED_HANDLER = (proxy, method, args) -> {
+        switch ( method.getName() ) {
+            case "close":
+                return Void.TYPE;
+            case "isClosed":
+                return Boolean.TRUE;
+            case "toString":
+                return CLOSED_HANDLER_STRING;
+            default:
+                throw new SQLException( "XAConnection for the XAResource is closed" );
+        }
+    };
+
+    private static final XAResource CLOSED_XA_RESOURCE = (XAResource) newProxyInstance( XAResource.class.getClassLoader(), new Class[]{XAResource.class}, CLOSED_HANDLER );
+
+    // --- //
+
+    private final ConnectionHandler handler;
+    private XAResource wrappedXAResource;
+
+    public XAResourceWrapper(ConnectionHandler connectionHandler, XAResource resource, boolean trackJdbcResources, AutoCloseableElement head) {
```

Comment:
Did you add trackJdbcResources for potential future usage?

Reply:
yes, and to pass it down to the `ConnectionWrapper` when `getConnection()` is invoked

Reply:
but there is no getConnection on [XAResource](https://docs.oracle.com/en/java/javase/21/docs/api/java.transaction.xa/javax/transaction/xa/XAResource.html)?

Reply:
you're right. I was confusing with XAConnection.